### ISSUE TITLE
Error when installing sparklyr if dplyr was not installed first

### ISSUE THREAD

**Issue by dfalty** [rstudio/sparklyr#47]
Windows 10
v0.99.1243
sparklyr 0.2.7

Error when installing sparklyr if dplyr was not installed first. Installation works fine if tibble or dplyr are installed first.

Steps:
1. Install sparklyr with not having dplyr installed
`devtools::install_github("rstudio/sparklyr")
2. Observe the following error
```
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
there is no package called 'tibble' 
ERROR: lazy loading failed for package 'sparklyr' 
* removing 'C:/Users/dfalt/Documents/R/win-library/3.3/sparklyr'
```

**Comment by kevinushey** [comment]
Very strange; `tibble` should be resolved in the dependency chain for `dplyr`. Are you still able to reproduce this? If so, we might just need to import `tibble` explicitly.

**Comment by dfalty** [comment]
Yeah, I'm still getting this on the latest.

**Comment by javierluraschi** [comment]
A mail thread with a customer described this problem as well. Did not spent time getting to the root cause of this issue since the workaround is safe enough to implement. Importing `tibble` explicitly now.

**Comment by dfalty** [comment]
Just FYI, the workaround works for me too on the latest. :)

**Comment by GriffinRidgeback** [comment]
This is not working for me.  I am installing from devtools and keep getting the following error:* installing *source* package ‘sparklyr’ ...
** R
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
Error : object 'db_data_type' not found whilst loading namespace 'sparklyr'
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Library/Frameworks/R.framework/Versions/3.3/Resources/library/sparklyr’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.3/Resources/library/sparklyr’

**Comment by sumendar** [comment]
Same error i got while installing sparklyr, any advice please.

**Comment by chrislarsenqlik** [comment]
I'm having the same issue with no package called 'glue' :(

**Comment by kevinykuo** [comment]
I've found that some installation related headaches are due to having an old R version. Might not solve everyone's problems but try updating R if you haven't in a while.

**Comment by jcorrean** [comment]
@kevinykuo I am facing the same problem even though I am working with the latest R release... Here's my session info
R version 3.4.3 (2017-11-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.3 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=es_CO.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=es_CO.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=es_CO.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=es_CO.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

loaded via a namespace (and not attached):
[1] compiler_3.4.3 tools_3.4.3    yaml_2.1.15

**Comment by mollenburger** [comment]
I had this same problem while trying to install dplyr in R 3.3 on MacOS X Sierra. 
I got errors when trying to install dplyr, the installer needed to update tibble from 1.3.0 to 1.3.4, and installing from source wasn't working.

I uninstalled tibble and tried reinstalling it (several times, both from CRAN, from local binary, from local source, restarting R in between), but each time I got a "Error: there is no package 'tibble'" ....while trying to install tibble. And since apparently ggplot2 depends on tibble I couldn't even just give up.

Updating R from 3.3 to 3.4 solved the problem...an hour later...
